packages/client/src/v2-events/features/events/AdvancedSearch/TabSearch.tsx

```diff
@@ -93,52 +93,69 @@ function enhanceEventFieldsWithSearchFieldConfig(event: EventConfig) {
   }
 }
 
-function getSectionFields(
-  event: EventConfig,
-  handleFieldChange: (fieldId: string, value: FieldValue) => void,
-  intl: IntlShape,
+function SectionFields({
+  event,
+  handleFieldChange,
+  fieldValues
+}: {
+  event: EventConfig
+  handleFieldChange: (fieldId: string, value: FieldValue) => void
   fieldValues?: Record<string, string>
-) {
+}) {
+  const intl = useIntl()
   const advancedSearchSections = event.advancedSearch
   const allUniqueFields = getAllUniqueFields(event)
-  return advancedSearchSections.map((section) => {
-    const metadataFields = getDefaultSearchFields(section)
-    const advancedSearchFieldId = section.fields.map(
-      (f: { fieldId: string }) => f.fieldId
-    )
-    const advancedSearchFields: Inferred[] = allUniqueFields.filter(
-      (field: Inferred) => advancedSearchFieldId.includes(field.id)
-    )
-
-    const combinedFields = [...metadataFields, ...advancedSearchFields]
-    const modifiedFields = combinedFields.map((f) => ({
-      ...f,
-      required: false as const, // advanced search fields need not be required
-      defaultValue: undefined // advanced search fields need no default or initial value
-    }))
 
-    return (
-      <Accordion
-        key={section.title.id}
-        expand={true}
-        label={intl.formatMessage(section.title)}
-        labelForHideAction={intl.formatMessage(messages.hide)}
-        labelForShowAction={intl.formatMessage(messages.show)}
-        name={section.title.id}
-      >
-        <FormFieldGenerator
-          fields={modifiedFields}
-          id={section.title.id}
-          initialValues={fieldValues}
-          onChange={(updatedValues) => {
-            Object.entries(updatedValues).forEach(([fieldId, value]) => {
-              handleFieldChange(fieldId, value)
-            })
-          }}
-        />
-      </Accordion>
-    )
-  })
+  return (
+    <>
+      {advancedSearchSections.map((section) => {
+        const metadataFields = getDefaultSearchFields(section)
+        const advancedSearchFieldId = section.fields.map(
+          (f: { fieldId: string }) => f.fieldId
+        )
+        const advancedSearchFields: Inferred[] = allUniqueFields.filter(
+          (field: Inferred) => advancedSearchFieldId.includes(field.id)
+        )
+
+        const combinedFields = [...metadataFields, ...advancedSearchFields]
+        const modifiedFields = combinedFields.map((f) => {
+          const fieldSearchConfig = advancedSearchSections
+            .flatMap((a) => a.fields)
+            .find((a) => a.fieldId === f.id)
```

Comment:
Do we need all the section fields here or would the current section fields suffice?